From 894221bded104a30034e6b1f7f6dae7fc998720b Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 7 Nov 2022 22:04:54 +0000
Subject: [PATCH 6/9] gnu: python-asdf: Update to 2.13.0

* gnu/packages/astronomy.scm (python-asdf): Update to 2.13.0
  [build-system]: Use pyproject-build-system
  [native-inputs]: Add python-astropy, python-psutil, python-pytest,
  python-pytest-doctestplus, python-pytest-openfiles,
  python-pytest-remotedata, python-semantic-version
  [propagated-inputs]: Add python-asdf-standard,
  python-asdf-transform-schemas, python-asdf-unit-schemas
  python-importlib-metadata, python-jsonschema-next, python-lz4.
  Remove python-jsonschema.
---
 gnu/packages/astronomy.scm | 26 +++++++++++++++++++-------
 1 file changed, 19 insertions(+), 7 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index a8c51338ac..a9bd687e66 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -55,6 +55,7 @@ (define-module (gnu packages astronomy)
   #:use-module (gnu packages python)
   #:use-module (gnu packages python-build)
   #:use-module (gnu packages python-check)
+  #:use-module (gnu packages python-compression)
   #:use-module (gnu packages python-crypto)
   #:use-module (gnu packages python-science)
   #:use-module (gnu packages python-web)
@@ -1916,26 +1917,37 @@ (define-public python-suntime
 (define-public python-asdf
   (package
     (name "python-asdf")
-    (version "2.8.3")
+    (version "2.13.0")
     (source
      (origin
        (method url-fetch)
        (uri (pypi-uri "asdf" version))
        (sha256
-        (base32 "0i4vq1hsympjgb1yvn4ql0gm8j1mki9ggmj03533kmg0nbzp03yy"))))
-    (build-system python-build-system)
+        (base32 "1zixzv4n2fryaszsfchqh2nvp0gzvarhz03fc721yw6iafdadqij"))))
+    (build-system pyproject-build-system)
     (arguments
      ;; NOTE: (Sharlatan-20211229T201059+0000): Tests depend on astropy and
      ;; gwcs, astropy gwcs depend on asdf.  Disable circular dependence.
      `(#:tests? #f))
     (native-inputs
-     (list python-setuptools-scm
+     (list python-astropy
+           python-packaging
+           python-psutil
+           python-pytest
+           python-pytest-doctestplus
+           python-pytest-openfiles
+           python-pytest-remotedata
            python-semantic-version
-           python-packaging))
+           python-setuptools-scm))
     (propagated-inputs
-     (list python-importlib-resources
-           python-jsonschema
+     (list python-asdf-standard
+           python-asdf-transform-schemas
+           python-asdf-unit-schemas
+           python-importlib-metadata
+           python-importlib-resources
            python-jmespath
+           python-jsonschema-next
+           python-lz4
            python-numpy
            python-pyyaml))
     (home-page "https://github.com/asdf-format/asdf")
-- 
2.38.0

